### PR TITLE
[Core][Hotfix] Using pointer/reference for variables on IntegrationValuesExtrapolationToNodesProcess 

### LINKED ISSUE #4483: Both VTK and GiD output plotted for all examples
i think we shouldn't output both. No problem in SPECIFYING to have both inputs...but this should not be the default, otherwise we double the disk occupation

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #4483

### CODE REVIEW

File: kratos/processes/integration_values_extrapolation_to_nodes_process.cpp [hunk 1]
```diff
@@ -43,26 +43,26 @@ IntegrationValuesExtrapolationToNodesProcess::IntegrationValuesExtrapolationToNo
...
-            mVectorVariable.push_back(variable);
-        } else if (KratosComponents< Variable< Matrix > >::Has(variable_name)) {
-            Variable<Matrix> variable = KratosComponents< Variable<Matrix> >::Get(variable_name);
-            mMatrixVariable.push_back(variable);
+    for (IndexType p_var = 0; p_var < n_variables; ++p_var){
+        const std::string& r_variable_name = ThisParameters["list_of_variables"].GetArrayItem(p_var).GetString();
+
+        if (KratosComponents<Variable<double>>::Has(r_variable_name)){
+            const Variable<double>& r_variable = KratosComponents< Variable<double>>::Get(r_variable_name);
+            mDoubleVariable.push_back(&(const_cast<Variable<double>&>(r_variable)));
```
Comment (philbucher) [inline kratos/processes/integration_values_extrapolation_to_nodes_process.cpp:56]: why extracting as const and then const_cast?

cannot you do `mDoubleVariable.push_back(KratosComponents< Variable<double>>::Get(r_variable_name));`?

Comment (loumalouomega) [reply kratos/processes/integration_values_extrapolation_to_nodes_process.cpp:56]: That means your vector is fill with constant references, you cannot declare a vector of constant references, you can do it with pointers

Comment (philbucher) [reply kratos/processes/integration_values_extrapolation_to_nodes_process.cpp:56]: `mDoubleVariable.push_back(*(KratosComponents< Variable<double>>::Get(r_variable_name)));` then?

Comment (philbucher) [reply kratos/processes/integration_values_extrapolation_to_nodes_process.cpp:56]: what I don't get is why you get as const and then const_cast in the next line

Comment (loumalouomega) [reply kratos/processes/integration_values_extrapolation_to_nodes_process.cpp:56]: Does not compile (cannot convert const Variable<double>& to Variable<double>* without removing const). I think in this case is safe to drop the const qualifier, as long as we only use the variables, we don't change them

Comment (loumalouomega) [reply kratos/processes/integration_values_extrapolation_to_nodes_process.cpp:56]: > what I don't get is why you get as const and then const_cast in the next line

To convert it into a pointer, I can do it all at the same time, but the line gets quite long...

Comment (philbucher) [reply kratos/processes/integration_values_extrapolation_to_nodes_process.cpp:56]: how abt the following then?
~~~c++
auto& r_variable = KratosComponents< Variable<double>>::Get(r_variable_name);
mDoubleVariable.push_back(&r_variable);
~~~

Comment (loumalouomega) [reply kratos/processes/integration_values_extrapolation_to_nodes_process.cpp:56]: Does not work with dropping the const (I already tried)

Comment (philbucher) [reply kratos/processes/integration_values_extrapolation_to_nodes_process.cpp:56]: ok it really does not work :)

**Comment by philbucher** [comment]
referencing #3789 for future reference, maybe a better solution can be found in the future

**Review by philbucher — COMMENTED** [review]
thanks for the fast fix

I tired it with the case posted in #4483, works without segfault now

**Comment by loumalouomega** [comment]
Can you approve then?

**Review by philbucher — APPROVED** [review]
missed the approve option :)

**Pull request merged**